pkg/openlaneclient/options.go

```diff
@@ -77,16 +77,9 @@ func WithTransport(transport http.RoundTripper) ClientOption {
 // WithCSRFToken sets the CSRF token header on the client for all requests
 func WithCSRFToken(token string) ClientOption {
 	return func(c *APIv1) error {
-		if token == "" {
-			return nil
-		}
-
-		if c.Requester.Header == nil {
-			c.Requester.Header = make(http.Header)
-		}
+		c.Config.Interceptors = append(c.Config.Interceptors, WithCSRFTokenInterceptor(token))
```

Comment:
I don't think you should remove the early return here so the header and interceptor are only applied when a non-empty token is provided